oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -384,8 +361,12 @@ pub fn wait_on_channels(
         }
         let statuses = readers_statuses(readers);
 
-        if statuses.iter().all(|s| s.is_err()) || statuses.iter().any(|s| s.unwrap_or(false)) {
-            return statuses;
+        if statuses
```

Comment:
It may be slightly more readable to name the intermediate clauses, e.g.

```rust
let all_unreadable = ...
let at_least_one_ready = ...
if all_unreadable || at _least_one_ready { ... }
```

but up to you.

Reply:
Done